lib/collection/src/shards/local_shard/clock_map.rs

```diff
@@ -294,3 +334,38 @@ impl From<Error> for CollectionError {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_clock_map_serde_empty() {
+        let clock_map = ClockMap::default();
+
+        let temp_dir = tempfile::tempdir().unwrap();
+        let path = temp_dir.path().join("clock_map_empty.json");
+
+        clock_map.store(&path).unwrap();
+        let clock_map2 = ClockMap::load(&path).unwrap();
+
+        assert_eq!(clock_map, clock_map2);
+    }
+
+    #[test]
+    fn test_clock_map_serde() {
+        let mut clock_map = ClockMap::default();
+        clock_map.advance_clock(ClockTag::new(1, 1, 1));
+        clock_map.advance_clock(ClockTag::new(1, 2, 8));
+        clock_map.advance_clock(ClockTag::new(2, 1, 42));
+        clock_map.advance_clock(ClockTag::new(2, 2, 12345));
+
+        let temp_dir = tempfile::tempdir().unwrap();
+        let path = temp_dir.path().join("clock_map.json");
+
+        clock_map.store(&path).unwrap();
+        let clock_map2 = ClockMap::load(&path).unwrap();
+
+        assert_eq!(clock_map, clock_map2);
+    }
```

Comment:
Do you mind if I switch to `serde_json::to_value`/`serde_json::from_value` to not dump stuff onto disk in these tests?

Reply:
Sure, good idea. :+1: